src/app/pipes/product-search.pipe.spec.ts

```diff
@@ -0,0 +1,54 @@
+import { Component } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { Product } from '../models/product';
+import { ProductSearchPipe } from './product-search.pipe';
+
+@Component({
+  selector: 'tk-test',
+  template:
+    '<span *ngFor="let product of products | productSearch: search"></span>'
+})
+class TestComponent {
+  products: ReadonlyArray<Product>;
+  search: string;
+}
+
+describe('AppModule => ProductSearchPipe', () => {
+  let fixture: ComponentFixture<TestComponent>;
+  let component: TestComponent;
+  const products = [{ name: 'first' }, { name: 'second' }] as Product[];
```

Comment:
It is not recommended to use **_as_** casting, even if the Product interface does not have *name* property, this code won't fail which is bad.